labs/6.md

```diff
@@ -1,72 +1,160 @@
+the [user guide](https://github.com/kubernetes-sigs/kubefed/blob/master/docs/userguide.md).
 
-NOTE: The version may change as the operator matures. Verify that the version of
-Federation matches the version of `kubefedctl`.
 
+The first step is to clone the demo code to your local machine:
 ~~~sh
-curl -LOs https://github.com/kubernetes-sigs/kubefed/releases/download/v0.0.10/kubefedctl.tgz
-tar xzf kubefedctl.tgz -C ~/bin
-rm -f kubefedctl.tgz
+git clone https://github.com/openshift/federation-dev.git
+cd federation-dev/federated-mongodb/mongo-yaml
 ~~~
 
-Verify that `kubefedctl` is working:
+Modify the cluster names to reflect the names of the kubefedclusters defined above.
 ~~~sh
-kubefedctl version
-
-kubefedctl version: version.Info{Version:"v0.0.10-dirty", GitCommit:"71d233ede685707df554ef653e06bf7f0229415c", GitTreeState:"dirty", BuildDate:"2019-05-06T22:30:31Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
+sed -i 's/feddemocl1/east1/g' ./*.yaml
+sed -i 's/feddemocl2/east2/g' ./*.yaml
+sed -i 's/feddemocl3/west2/g' ./*.yaml
 ~~~
 
-## Joining Clusters
-Now that the `kubefedctl` binary has been acquired the next step is joining the clusters.
-`kubefedctl` binary utilizes the contexts and clusters within `kubeconfig` when defining the clusters.
-
-Using the `kubeconfig` file that was generated, verify the Operator has been successfully deployed.
+Before deploying MongoDB the yaml files need to be updated to define the certificates that
+were created as well as the routing endpoints that will be used. Ensure the values of `mongo.pem` and `ca.pem` reflect the path where the `pem` files were created.
 ~~~sh
-$ oc get csv -n federated-mongo
-NAME                DISPLAY      VERSION   REPLACES   PHASE
-federation.v0.0.10   Federation   0.0.10                Succeeded
+# Define the pem
+sed -i "s/mongodb.pem: .*$/mongodb.pem: $(openssl base64 -A < mongo.pem)/" 01-mongo-federated-secret.yaml
+sed -i "s/ca.pem: .*$/ca.pem: $(openssl base64 -A < ca.pem)/" 01-mongo-federated-secret.yaml
+# Configure MongoDB Endpoints for the deployment
+sed -i "s/primarynodehere/${ROUTE_CLUSTER1}:443/" 04-mongo-federated-deployment-rs.yaml
+sed -i "s/replicamembershere/${ROUTE_CLUSTER1}:443,${ROUTE_CLUSTER2}:443,${ROUTE_CLUSTER3}:443/" 04-mongo-federated-deployment-rs.yaml
 ~~~
-The next step is to federate the clusters using `kubefedctl`.
+
+Now that the yaml files contain the `pem` and routes it is time to deploy the objects.
 ~~~sh
-kubefedctl join east1 --host-cluster-context east1 --add-to-registry --v=2 --federation-namespace=federated-mongo
-kubefedctl join east2 --host-cluster-context east1 --add-to-registry --v=2 --federation-namespace=federated-mongo
-kubefedctl join west2 --host-cluster-context east1 --add-to-registry --v=2 --federation-namespace=federated-mongo
+# Create secret with admin/access credentials
+oc --context=east1 -n federated-mongo create -f 01-mongo-federated-secret.yaml
```

Comment:
will these be the actual contexts we use in the lab?  east1, west2, etc..?

Reply:
should we rename them "cluster1", and "cluster2" or something they may grok easier?